### ISSUE TITLE
Questions reagarding adjacency matrix and loss

### ISSUE THREAD

**Issue by aktersnurra** [abduallahmohamed/Social-STGCNN#39]
Hi,

Interesting project, but I would like to ask a couple of questions regarding the adjacency matrix and loss calculation that are unclear to me.

Why does the adjacency matrix not have a batch dimension? I thought it was dependent on the nodes in a scene(?)

https://github.com/abduallahmohamed/Social-STGCNN/blob/dbbb111a0f645e4002dd4885564da226c5e0b19f/model.py#L67

Could you also explain the if else statement when computing the loss, I find it rather confusing.

https://github.com/abduallahmohamed/Social-STGCNN/blob/dbbb111a0f645e4002dd4885564da226c5e0b19f/train.py#L178

Cheers,

aktersnurra

**Comment by abduallahmohamed** [comment]
Hi, 
Thanks for asking. 
for _Why does the adjacency matrix not have a batch dimension? I thought it was dependent on the nodes in a scene(?)_
The graph are dynamic in size, in a scene there could be 3 pedestrians, in another it could be 30 pedestrians. Thus batching in our case requires a specific implementation which is not available for now.
for _Could you also explain the if else statement when computing the loss, I find it rather confusing._
Also because batching isn't available in our case, in order to have a faster training, I backprop on each 128 individually passed batches. The turn_point variable is just an indicator for the last point in a non complete 128 samples, aka the fractions after taking a group of 128 samples.

**Comment by aktersnurra** [comment]
Hi Mohamed, 

Ok, I understand. Thanks for the fast response! 

Again, really cool work you guys have done. 

Thanks for sharing!

Cheers,

aktersnurra
